mlfinlab/backtest_statistics/backtests.py

```diff
@@ -0,0 +1,546 @@
+    """
+
+    def __init__(self, simulations=2000):
+        """
+        Set the desired number of simulations to make in Haircut Sharpe Ratios or Profit Hurdle algorithms.
+
+        :param simulations: (int) Number of simulations
+        """
+
+        self.simulations = simulations
+
+    @staticmethod
+    def _sample_random_multest(rho, n_trails, prob_zero_mean, lambd, n_simulations, annual_vol=0.15, n_obs=240):
+        """
+        Generates empirical p-value distributions.
+
+        The algorithm is described in the paper and is based on the model estimated by `Harvey, C.R., Y. Liu,
+        and H. Zhu., … and the Cross-section of Expected Returns. Review of Financial Studies, forthcoming 2015`,
+        referred to as the HLZ model.
+
+        It provides a set of simulated t-statistics based on the parameters recieved from the _parameter_calculation
+        method.
+
+        Researchers propose a structural model to capture trading strategies’ underlying distribution.
+        With probability p0 (prob_zero_mean), a strategy has a mean return of zero and therefore comes
+        from the null distribution. With probability 1 – p0, a strategy has a nonzero mean and therefore
+        comes from the alternative distribution - exponential.
+
+        :param rho: (float) Average correlation among returns
+        :param n_trails: (int) Total number of trials inside a simulation
+        :param prob_zero_mean: (float) Probability for a random factor to have a zero mean
+        :param lambd: (float) Average of monthly mean returns for true strategies
+        :param n_simulations: (int) Number of rows (simulations)
+        :param annual_vol: (float) HLZ assume that the innovations in returns follow a normal distribution with a mean
+                                   of zero and a standard deviation of ma = 15%
+        :param n_obs: (int) Number of observations of used for volatility estimation from HLZ
+        :return: (np.ndarray) Array with distributions calculated
+        """
+
+        # Assumed level of monthly volatility = adjusted yearly volatility
+        monthly_volatility = annual_vol / 12 ** (1 / 2)
+
+        # Creating a correlation matrix of simulated returns. All correlations are assumed to be the same as average
+        # correlation among returns
+        # The first row of the correlation matrix: [1, rho, rho, .., rho]
+        correlation_vector = np.insert(rho * np.ones((1, n_trails - 1)), 0, 1)
+        # Correlation matrix created from the vector by expanding it
+        correlation_matrix = linalg.toeplitz(correlation_vector)
+        # Vector with mean of simulated returns - zeros
+        mean = np.zeros(n_trails)
+
+        # Creating a sample from a multivariate normal distribution as returns simulations
+        # Covariance matrix - Created from correlation matrix multiplied by monthly volatility and adjusted
+        covariance_matrix = correlation_matrix * (monthly_volatility ** 2 / n_obs)
```

Comment:
Please remove spaces between comments. That way it looks easy to read the code. Do this for all parts of the code. Something like this:

```
# Creating a sample from a multivariate normal distribution as returns simulations
# Covariance matrix - Created from correlation matrix multiplied by monthly volatility and adjusted
covariance_matrix = correlation_matrix * (monthly_volatility ** 2 / n_obs)

 # Result - n_simulations rows with n_trails inside
 shock_mat = np.random.multivariate_normal(mean, covariance_matrix, n_simulations)
```

Reply:
Will do